web/src/components/eventdetail/schedules.js

```diff
@@ -0,0 +1,39 @@
+import {
+  Typography,
+} from '@material-ui/core';
+import PropTypes from 'prop-types';
+import React, {Component} from 'react';
+
+class Schedules extends Component {
+  constructor(props) {
+    super(props);
+  }
+
+  render() {
+    const scheduleList = [];
+    let scheduleLength = 3;
```

Comment:
if (length >= 3 || !this.props.moreButton), this code makes scheduleLength = 3. This will then only go over up to 3 elements down in the loop. Is this intentional? 

Also, the 2 blocks in if / else if do the same thing (line 16, 18). 

It's weird to include moreButton into the logic. Instead, just rely on children.length to do the rendering, and add a test to see if we're correctly adding moreButton:

scheduleLength = Math.min(this.props.children.length, 3);  // '3' should be declared elsewhere.

Reply:
My intention is if there are more than 3 schedules, page just display 3 schedules with more button at first. And if user clicked 'more button', page display all of schedules. So I used 'moreButton' to check button's status.

Reply:
Now, I just fixed `scheduleLength = Math.min(this.props.children.length, 3);` and deleted line 17-18.